lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet.py

```diff
@@ -15,193 +15,351 @@
+    type: int
     default: 120
   backups:
     description:
      - indicates whether automated backups should be enabled.
-    required: False
     default: False
     type: bool
   monitoring:
     description:
-     - indicates whether to install the DigitalOcean agent for monitoring.
-    required: False
+     - Install DigitalOcean monitoring agent on a new Droplet if C(True).
     default: False
     type: bool
   tags:
     description:
-     - List, A list of tag names as strings to apply to the Droplet after it is created. Tag names can either be existing or new tags.
-    required: False
+     - A list of strings to tag the Droplet with on creation. A tag can be existing or new.
+    type: list
   volumes:
     description:
-     - List, A list including the unique string identifier for each Block Storage volume to be attached to the Droplet.
-    required: False
-  oauth_token:
+     - A list of Block Storage volume identifiers to attach to new Droplet.
+     - "Note: a volume can only be attached to a single Droplet."
+    type: list
+  rebuild:
     description:
-     - DigitalOcean OAuth token. Can be specified in C(DO_API_KEY), C(DO_API_TOKEN), or C(DO_OAUTH_TOKEN) environment variables
-    aliases: ['API_TOKEN']
-    required: True
+     - force Droplet rebuild. You may supply I(image) id/slug if you want it changed or omit I(image) and just re-fresh your Droplet.
+     - 'WARNING: droplet own data will be lost!'
+    default: False
+    type: bool
 requirements:
   - "python >= 2.6"
+
+extends_documentation_fragment: digital_ocean.documentation
 '''
 
 
 EXAMPLES = '''
-- name: create a new droplet
+- name: create new droplet. If droplet with this name exists other parameters ignored.
   digital_ocean_droplet:
     state: present
-    name: mydroplet
-    oauth_token: XXX
-    size: 2gb
+    name: mydroplet.example.com
+    oauth_token: "{{ lookup('file', '~/.do/api-key1') }}"
+    size: 1gb
     region: sfo1
     image: ubuntu-16-04-x64
+    tags: [ 'foo', 'bar' ]
     wait_timeout: 500
   register: my_droplet
 
 - debug:
     msg: "ID is {{ my_droplet.data.droplet.id }}, IP is {{ my_droplet.data.ip_address }}"
 
-- name: ensure a droplet is present
+- name: Check droplet exists, get details
   digital_ocean_droplet:
-    state: present
-    id: 123
-    name: mydroplet
-    oauth_token: XXX
-    size: 2gb
-    region: sfo1
-    image: ubuntu-16-04-x64
-    wait_timeout: 500
+    name: mydroplet.example.com
+    oauth_token: "{{ lookup('file', '~/.do/api-key1') }}"
+  register: my_droplet
+
+- name: ensure a droplet is like new
+  digital_ocean_droplet:
+    name: mydroplet.example.com
+    oauth_token: "{{ lookup('file', '~/.do/api-key1') }}"
+    rebuild: yes
+    image: debian-9-x64  # may be omitted.
+    wait_step: 3.5
+    wait_build: 16  # wait 16 seconds, then GET DO API droplet status with 3.5 seconds pauses, unless default wait_timeout (120 s) is reached.
```

Comment:
Rename DO to DigitalOcean to avoid acronyms

Reply:
3 DO occurrences replaced as you suggested.